### ISSUE TITLE
MobileServiceInvalidOperationException: 500 Error

### ISSUE THREAD

**Issue by ericbrunner** [Azure/azure-mobile-apps#47]
**Nuget Versions:**

```
Microsoft.Azure.Mobile.Client: v.3.1.0
Microsoft.Azure.Mobile.Server: v.2.0.0
```

Hi,

I recently get some HTTP 500 (Internal Server Error) Error when my Mobile App requests REST API operations in my Mobile Apps backend.

I turned off customErrors in my bakcend to see the exact root cause of the `MobileServiceInvalidOperationException` exception I get transferred to 
my Xamarin.Forms mobile App. Below is the response. Not expressive at all!

<!DOCTYPE html> <html> <head> <title>Cannot access a disposed object.<br>Object name: 'System.Net.Http.HttpMessageInvoker'.</title> <meta name='viewport' content='width=device-width' /> <style> body {font-family:'Verdana';font-weight:normal;font-size: .7em;color:black;} p {font-family:'Verdana';font-weight:normal;color:black;margin-top: -5px} b {font-family:'Verdana';font-weight:bold;color:black;margin-top: -5px} H1 { font-family:'Verdana';font-weight:normal;font-size:18pt;color:red } H2 { font-family:'Verdana';font-weight:normal;font-size:14pt;color:maroon } pre {font-family:'Consolas','Lucida Console',Monospace;font-size:11pt;margin:0;padding:0.5em;line-height:14pt} .marker {font-weight: bold; color: black;text-decoration: none;} .version {color: gray;} .error {margin-bottom: 10px;} .expandable { text-decoration:underline; font-weight:bold; color:navy; cursor:hand; } @media screen and (max-width: 639px) { pre { width: 440px; overflow: auto; white-space: pre-wrap; word-wrap: break-word; } } @media screen and (max-width: 479px) { pre { width: 280px; } } </style> </head> <body bgcolor='white'> <span><H1>Server Error in '/' Application.<hr width=100% size=1 color=silver></H1> <h2> <i>Cannot access a disposed object.<br>Object name: 'System.Net.Http.HttpMessageInvoker'.</i> </h2></span> <font face='Arial, Helvetica, Geneva, SunSans-Regular, sans-serif '> <b> Description: </b>An unhandled exception occurred during the execution of the current web request. Please review the stack trace for more information about the error and where it originated in the code. <br><br> <b> Exception Details: </b>System.ObjectDisposedException: Cannot access a disposed object.<br>Object name: 'System.Net.Http.HttpMessageInvoker'.<br><br> <b>Source Error:</b> <br><br> <table width=100% bgcolor='#ffffcc'> <tr> <td> <code> An unhandled exception was generated during the execution of the current web request. Information regarding the origin and location of the exception can be identified using the exception stack trace below.</code> </td> </tr> </table> <br> <b>Stack Trace:</b> <br><br> <table width=100% bgcolor='#ffffcc'> <tr> <td> <code><pre> [ObjectDisposedException: Cannot access a disposed object. Object name: &#39;System.Net.Http.HttpMessageInvoker&#39;.] System.Net.Http.HttpMessageInvoker.CheckDisposed() +327240 System.Net.Http.HttpMessageInvoker.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) +24 System.Web.Http.Owin.&lt;InvokeCore&gt;d__0.MoveNext() +496 System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) +92 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) +58 Microsoft.Owin.Host.SystemWeb.IntegratedPipeline.&lt;RunApp&gt;d__5.MoveNext() +182 System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) +92 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) +58 Microsoft.Owin.Host.SystemWeb.IntegratedPipeline.&lt;DoFinalWork&gt;d__2.MoveNext() +180 Microsoft.Owin.Host.SystemWeb.IntegratedPipeline.StageAsyncResult.End(IAsyncResult ar) +69 Microsoft.Owin.Host.SystemWeb.IntegratedPipeline.IntegratedPipelineContext.EndFinalWork(IAsyncResult ar) +64 System.Web.AsyncEventExecutionStep.System.Web.HttpApplication.IExecutionStep.Execute() +380 System.Web.HttpApplication.ExecuteStep(IExecutionStep step, Boolean&amp; completedSynchronously) +155 </pre></code> </td> </tr> </table> <br> <hr width=100% size=1 color=silver> <b>Version Information:</b>&nbsp;Microsoft .NET Framework Version:4.0.30319; ASP.NET Version:4.6.1590.0 </font> </body> </html> 

It happens at as response to PATCH and GET request like them:

`PATCH /tables/TruckAuftragWorkFlow/8BB91730-98A7-4A3A-9BB2-494EB9253AD2`

`GET /tables/TruckWorkFlowStamm`

`https://trucker-rolsped.azurewebsites.net/tables/TruckWorkFlowStamm?$filter=(updatedAt ge datetimeoffset'2017-05-24T14:40:43.2200000%2B00:00')&$orderby=updatedAt&$skip=1&$top=50&__includeDeleted=true`

**Questions:**

1. Why does that happen from time to time? 
3. Is there some bugfix (Azure Mobile App Service client , server) ?

thanks,
Eric

**Comment by elamalani** [comment]
@ericbrunner Do you still see this error? Our latest NuGet release is v4.0.2 if you are not on it already  - https://github.com/Azure/azure-mobile-apps-net-client/releases/tag/4.0.2

Please keep us posted. Thanks!

**Comment by ericbrunner** [comment]
To be honest, I have to investigate. I posted that issue on on 8 Jun 2017 which is **10 month ago**. So if that is that support Microsoft provides I stop posting any issue and use other stuff.

But anyway: I looked into the AI logs and didn't find any of those Exceptions recently. I expect that as a Azure Portal App Service Hosting issue that obviously got fixed because I still use the same Client / Server version.

Nuget Versions:

```
Microsoft.Azure.Mobile.Client: v.3.1.0
Microsoft.Azure.Mobile.Server: v.2.0.0
```
